pkg/providers/vsphere/vsphere_test.go

```diff
@@ -429,17 +430,35 @@ func newProviderWithGovc(t *testing.T, datacenterConfig *v1alpha1.VSphereDatacen
 
 func newProvider(t *testing.T, datacenterConfig *v1alpha1.VSphereDatacenterConfig, machineConfigs map[string]*v1alpha1.VSphereMachineConfig, clusterConfig *v1alpha1.Cluster, govc ProviderGovcClient, kubectl ProviderKubectlClient, resourceSetManager ClusterResourceSetManager) *vsphereProvider {
 	_, writer := test.NewWriter(t)
+	netClient := &DummyNetClient{}
+	ctrl := gomock.NewController(t)
+	vSphereClientCallable := func(ctx context.Context, host string, username string, password string, insecure bool, datacenter string) (VSphereClient, error) {
+		vsc := mocks.NewMockVSphereClient(ctrl)
+		vsc.EXPECT().Username().Return("foobar").AnyTimes()
+
+		var privs []string
+		err := json.Unmarshal([]byte(vsphereAdminPrivsFile), &privs)
+		if err != nil {
+			return nil, err
+		}
+
+		vsc.EXPECT().GetPrivsOnEntity(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(privs, nil).AnyTimes()
+		return vsc, nil
+	}
+	validator := NewValidator(govc, netClient, vSphereClientCallable)
```

Comment:
I feel like this code is messy. I needed a way to inject a dummy vsphere client into the validator for my tests. I am extremely open to alternative suggestions.